### PR TITLE
Add data source for compute keypair.

### LINKED ISSUE #306: Data source for keypair.
Hey there!

Would y'all accept a PR to add a data source for keypairs?

### PR DESCRIPTION
**Pull request by genevieve**
Fixes #306.

- Adds doc.
- Adds acceptance test.

### CODE REVIEW

**Comment by jtopjian** [comment]
@genevieve Really nice work with this.

I'm not sure what's causing the Travis failure. Let's ignore it for now.

The only thing missing is a sidebar entry here: https://github.com/terraform-providers/terraform-provider-openstack/blob/master/website/openstack.erb

File: website/docs/d/compute_keypair_v2.html.markdown [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  name = "sand"
+}
+```
+
+## Argument Reference
+
+* `region` - (Optional) The region in which to obtain the V2 Compute client.
+    If omitted, the `region` argument of the provider is used.
+
+* `name` - (Optional) The unique name of the keypair.
```
Comment (jtopjian) [inline website/docs/d/compute_keypair_v2.html.markdown]: Sorry, just spotted this. This should be Required.

**Comment by genevieve** [comment]
The error looks to be:
```
http://127.0.0.1:4567/docs/providers//:
Remote file does not exist -- broken link!!!
Found 1 broken link.
http://127.0.0.1:4567/docs/providers//
```

There must be a broken link in one of the docs changes I made?

**Comment by jtopjian** [comment]
> Edit: Actually it seems like the $URL passed to check-links.sh is missing the openstack provider name in the path?

Yes - exactly :)

It was an internal bug which has now been fixed.

**Comment by jtopjian** [comment]
This looks good to me. I'll let the openlab tests finish and then merge.

Thank you for working on this :)

**Comment by genevieve** [comment]
Thanks Joe!

**Pull request merged**